core/src/main/java/org/infinispan/remoting/transport/jgroups/JGroupsTransport.java

```diff
@@ -468,6 +476,7 @@ public void start() {
       setXSiteViewListener(channelCallbacks);
 
       startJGroupsChannelIfNeeded();
+      metricsRegistration.registerMetrics(channel, metricsCollector.wired());
```

Comment:
I think you should keep track of `MetricID` and unregister them on stop/disconnect.

Reply:
Done